### ISSUE TITLE
False positive for `file!()` macro in println

### ISSUE THREAD

**Issue by hellow554** [rust-lang/rust-clippy#10544]
### Summary

When using `file!()` as a macro inside of `println!`, it suggest to replace it.

The suggestion leads to invalid code.

The error was probably introduces in 6fc6d87fd0@Alexendoo 

As shown in the code below, other macro calls work fine, it seems to be a problem with the `file!()` macro in particular.

Maybe checking `&& value_string != "file!()"` would be a valid solution here?!
https://github.com/rust-lang/rust-clippy/blob/27edc2af066979ff7b3182b33c62a84d57b907c4/clippy_lints/src/write.rs#L463

### Lint Name

print_literal

### Reproducer

I tried this code:

```rust
macro_rules! static_str {
    () => { "Hello World" }
}

fn main() {
    println!("{}", file!());
    println!("{}", line!());
    println!("{}", static_str!());
    println!("{}", "Hello World");
}
```

I saw this happen:

```
warning: literal with an empty format string
 --> src/main.rs:6:20
  |
6 |     println!("{}", file!());
  |                    ^^^^^^^
  |
  = note: `#[warn(clippy::print_literal)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#print_literal
help: try this
  |
6 -     println!("{}", file!());
6 +     println!("ile!(");
  |

warning: literal with an empty format string
 --> src/main.rs:9:20
  |
9 |     println!("{}", "Hello World");
  |                    ^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#print_literal
help: try this
  |
9 -     println!("{}", "Hello World");
9 +     println!("Hello World");
  |

warning: `abc` (bin "abc") generated 2 warnings
```

I expected to see this happen:

Not issuing the first warning, but only the second

### Version

```text
rustc 1.66.0-nightly (f5193a9fc 2022-09-25)
binary: rustc
commit-hash: f5193a9fcc73dc09e41a90c5a2c97fc9acc16032
commit-date: 2022-09-25
host: x86_64-unknown-linux-gnu
release: 1.66.0-nightly
LLVM version: 15.0.0
```

### Additional Labels

@rustbot label +I-suggestion-causes-error

**Comment by samueltardieu** [comment]
The problem seems to be with the suggested replacement (which lacks a "f" at the beginning). Otherwise, using `println!(file!());` is fine, `file!()` is expanded into a string literal.
